### PR TITLE
Update the Ember example

### PR DESCRIPTION
**Pull request by Windvis**
The ember learning team prefers us not using undocumented / semi implemented / "modern ember" functionality until the release of the next Ember edition. This PR is only about updating the app to use the newest version of TodoMVC and display the sidebar with up-to-date information.

WIP until the Ember learning team approves of this change. I'll squash all commits, make a new build and update this PR after that.

### CODE REVIEW

File: examples/emberjs/todomvc/app/components/todo-list.js [hunk 1]
```diff
@@ -5,7 +5,7 @@ import Component from '@ember/component';
 export default Component.extend({
```
Comment (NullVoxPopuli) [inline examples/emberjs/todomvc/app/components/todo-list.js]: Should we be using es classes?

File: examples/emberjs/todomvc/app/templates/application.hbs [hunk 2]
```diff
@@ -1,24 +1,29 @@
...
-        </ul>
-        {{#if completed.length}}
-          <button id="clear-completed" onclick={{action 'clearCompleted'}}>Clear completed</button>
-        {{/if}}
-      </footer>
-    {{/if}}
+  {{outlet}}
+  {{#if (gt model.length 0)}}
+    <footer class="footer">
+      <span class="todo-count"><strong>{{remaining.length}}</strong> {{pluralize "item" remaining.length}} left</span>
```
Comment (NullVoxPopuli) [inline examples/emberjs/todomvc/app/templates/application.hbs]: I think all the variables and such should be prefixed with 'this'

Also, should we extract components?

File: examples/emberjs/todomvc/app/templates/components/todo-item.hbs [hunk 3]
```diff
@@ -1,6 +1,17 @@
 <div class="view">
-  <input type="checkbox" class="toggle" checked={{todo.completed}} onchange={{action 'toggleCompleted'}}>
-  <label ondblclick={{action 'startEditing'}}>{{todo.title}}</label>
-  <button onclick={{action 'removeTodo'}} class="destroy"></button>
+  <input
+    class="toggle"
+    type="checkbox"
+    checked={{todo.completed}}
```
Comment (NullVoxPopuli) [inline examples/emberjs/todomvc/app/templates/components/todo-item.hbs]: Maybe this.todo.isCompleted ?

File: examples/emberjs/todomvc/config/optional-features.json [hunk 4]
```diff
@@ -0,0 +1,3 @@
+{
+  "jquery-integration": true
```
Comment (NullVoxPopuli) [inline examples/emberjs/todomvc/config/optional-features.json]: Should be false, yeah?
We totes don't need jsquirrels

File: examples/emberjs/todomvc/app/templates/components/todo-list.hbs [hunk 5]
```diff
@@ -1,10 +1,21 @@
...
+      checked={{allCompleted}}
+      onchange={{action "toggleAll"}}
+    >
+    <label for="toggle-all">Mark all as complete</label>
   {{/if}}
-  <ul id="todo-list" class="todo-list">
+  <ul class="todo-list">
     {{#each todos as |todo|}}
-      {{todo-item todo=todo onStartEdit=(action 'disableToggle') onEndEdit=(action 'enableToggle')}}
+      {{todo-item
```
Comment (NullVoxPopuli) [inline examples/emberjs/todomvc/app/templates/components/todo-list.hbs]: Angle brackets can be used here :)

File: examples/emberjs/todomvc/app/components/todo-list.js (hunk 1 above)
Comment (NindroidX) [reply examples/emberjs/todomvc/app/components/todo-list.js]: Updated the PR description with more information.

File: examples/emberjs/todomvc/app/templates/application.hbs (hunk 2 above)
Comment (NindroidX) [reply examples/emberjs/todomvc/app/templates/application.hbs]: What components did you have in mind?

Comment (NullVoxPopuli) [reply examples/emberjs/todomvc/app/templates/application.hbs]: one for footer, and header :)

Comment (NindroidX) [reply examples/emberjs/todomvc/app/templates/application.hbs]: Decided against this for now. I think this MR should only about updating ember and use the newest TodoMVC template.

**Comment by NindroidX** [comment]
I squashed all commits and made a new production build. Should be good to merge after review.

**Comment by FadySamirSadek** [comment]
@NullVoxPopuli WDYT ?

**Comment by NullVoxPopuli** [comment]
It's def an improvement. We can iterate more after this is merged

**Comment by Windvis** [comment]
Thanks for merging this!

It can indeed be improved more, but I think updating the TodoMVC parts first was more important.

I also think that refactoring the code should be discussed with the learning team first. They seem to have a  specific mindset about these kind of projects (for example no fancy new Ember features until Octane lands). Maybe they want things to be "basic" to not overwhelm people interested?

**Comment by Windvis** [comment]
@FadySamirSadek When are these changes going live on the website? I didn't really find any information about when the website gets updated 😁.

**Pull request merged**